addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -2444,13 +2444,28 @@ var SnippetsMenu = Widget.extend({
         const $snippet = this.$snippets.has(`[data-snippet="${ev.data.key}"]`);
         ev.data.onSuccess($snippet.length ? $snippet[0].dataset.oeThumbnail : '');
     },
+    /**
+     * @private
+     */
+    _clearSelection(ev) {
+        if (window.getSelection) {
+            if (window.getSelection().empty) {  // Chrome
+                window.getSelection().empty();
+            } else if (window.getSelection().removeAllRanges) {  // Firefox
+                window.getSelection().removeAllRanges();
+            }
+        } else if (document.selection) {  // IE?
+            document.selection.empty();
+        }
+    },
     /**
      * @private
      */
     _onSummernoteToolsUpdate(ev) {
         if (!this._textToolsSwitchingEnabled) {
             return;
         }
+        this._clearSelection();
```

Comment:
Maybe the reported bug is fixed but now you cannot put your cursor anywhere in the page and type any text so obviously there is no more problem...

I don't think the fix is about changing the selection of the text once summernote detects a selection. Rather increasing the hacks in this `_onSummernoteToolsUpdate` method to not switch to the style tab when the selection is in an empty oe_structure or something like that?

Reply:
Hello @qsm-odoo Changes are done. Can you please check?
Thanks